client/my-sites/stats/stats-module/index.jsx

```diff
@@ -64,6 +64,25 @@ class StatsModule extends Component {
 		}
 	}
 
+	shouldComponentUpdate( nextProps ) {
+		for ( const key in nextProps ) {
+			if ( this.props[ key ] !== nextProps[ key ] ) {
+				if ( key !== 'data' ) {
+					return true; // found shallow unequal prop, should update.
+				}
+
+				// Special case for data prop, do deep equality check.
+				const isDeepEqual = isEqual( this.props[ key ], nextProps[ key ] );
```

Comment:
Some `data` payloads include nested child objects like Clicks, Referrers, and Authors - any concerns with using `isEqual` knowing that?  Specifically Referrers can potentially be 3 levels deep:

![stats_ _the_wordpress_com_blog_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/24626914/bc9094e2-1867-11e7-87cd-52b93f2bc29c.png)